src/lumigo_tracer/utils.py

```diff
@@ -138,22 +141,30 @@ def _create_request_body(
     msgs: List[dict],
     prune_size_flag: bool,
     max_size: int = MAX_SIZE_FOR_REQUEST,
-    min_size: int = MIN_SPAN_SIZE,
+    too_big_spans_threshold: int = TOO_BIG_SPANS_THRESHOLD,
 ) -> str:
-    if not prune_size_flag or _get_event_base64_size(msgs) < max_size:
+
+    if not prune_size_flag or (
+        len(msgs) < NUMBER_OF_SPANS_IN_REPORT_OPTIMIZATION
+        and _get_event_base64_size(msgs) < max_size  # noqa
+    ):
```

Comment:
Running only 1 time hereof `_get_event_base64_size(msgs)` is effecting so much ?

Reply:
hell yeah!
this depends on the number of spans. In my tests (3000 spans) it took a full second!